superset-frontend/src/filters/components/Time/TimeFilter.tsx

```diff
@@ -0,0 +1,72 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { styled, TimeRange } from '@superset-ui/core';
+import React, { useState, useEffect } from 'react';
+import DateFilterControl from 'src/explore/components/controls/DateFilterControl/DateFilterControl';
+import { getRangeExtraFormData } from 'src/filters/utils';
+import { AntdPluginFilterStylesProps } from '../types';
+import { DEFAULT_FORM_DATA, PluginFilterTimeProps } from './types';
+
+const Styles = styled.div<AntdPluginFilterStylesProps>`
+  height: ${({ height }) => height}px;
+  width: ${({ width }) => width}px;
+  overflow-x: scroll;
+`;
+
+export default function PluginFilterTime(props: PluginFilterTimeProps) {
+  const { formData, setExtraFormData, width, height } = props;
+  const { defaultValue } = {
+    ...DEFAULT_FORM_DATA,
+    ...formData,
+  };
+
+  const firstDefault = (defaultValue?.[0] || '').toString();
+  const [value, setValue] = useState<string>(firstDefault || 'Last week');
+
+  let { groupby = [] } = formData;
+  groupby = Array.isArray(groupby) ? groupby : [groupby];
+
+  const handleTimeRangeChange = (timeRange: TimeRange) => {
+    const [col] = groupby;
+    const extraFormData = getRangeExtraFormData(
+      col,
+      timeRange.since,
+      timeRange.until,
+    );
+    // @ts-ignore
+    setExtraFormData({ extraFormData, currentState: { value: [value] } });
+  };
```

Comment:
The handler should set `extraFormData` with the following params:
```javascript
extraFormData = {
  override: {
    time_range: timeRange,
  },
};
```

Reply:
I have changed it to:
```
    const { time_range } = timeRange;
    setExtraFormData({
      // @ts-ignore
      extraFormData: { override: { time_range } },
      currentState: { value: [time_range] },
    });

```
but it does not seem to work anymore

Reply:
Hmm, let me test this really quickly, it's probably something very simple that's wrong.

Reply:
I checked, this would be roughly the correct hook call (assuming `time_range` is what's generated by `buildTimeRangeString`):
```
    setExtraFormData({
      extraFormData: {
        override_form_data: {
          time_range,
        }
      },
      currentState: { value: [timeRange] },
    });
```
Btw, you'll need to `@ts-ignore` until we update the `ExtraFormData` type in `superset-ui/core`